source-postgres/fdbutils.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/estuary/flow/go/protocols/fdb/tuple"
+	"github.com/jackc/pgtype"
+)
+
+const numericKey = "N"
+
+// EncodePgNumericKeyFDB encodes a pgtype.Numeric value to an FDB tuple.
+func EncodePgNumericKeyFDB(key pgtype.Numeric) (tuple.Tuple, error) {
+	if encodedKey, err := encodePgNumeric(key); err != nil {
+		return nil, fmt.Errorf("encode pgtype.Numeric: %w", err)
+	} else if encodedBinary, err := key.EncodeBinary(nil, nil); err != nil {
+		return nil, fmt.Errorf("encode binary pgtype Numeric: %w", err)
+	} else {
+		return tuple.Tuple{numericKey, encodedKey, encodedBinary}, nil
+	}
+}
+
+// DecodeFDBTuple decodes the input TupleElement to its original value.
+// Return nil if the input cannot be decoded.
+func DecodeFDBTuple(t tuple.TupleElement) interface{} {
+	switch t := t.(type) {
+	case tuple.Tuple:
+		if len(t) == 3 {
+			if key, ok := t[0].(string); ok && key == numericKey {
```

Comment:
Style note: I'm pretty sure you can just write `t[0] == numericKey` to get the same logical behavior here.

I'd recommend combining the two levels of checks into something like `len(t) == 3 && t[0] == numericKey` because that makes the pattern more clear if/when we end up adding more custom encoding cases in the future.